Core/src/Geometry/ProtoLayer.cpp

```diff
@@ -35,14 +35,14 @@ ProtoLayer::ProtoLayer(
 
 double ProtoLayer::min(BinningValue bval, bool addenv) const {
   if (addenv) {
-    return extent.min(bval) - envelope[bval].first;
+    return extent.min(bval) - std::abs(envelope[bval][0u]);
```

Comment:
why is `abs` necessary here?

Reply:
Actually, to be completely save, it should also be applied at the `max()`, but I guess, what I will do is remove this, and added potentially a check.